train.py

```diff
@@ -169,13 +166,15 @@ def loss_fn(pred, labels):
             else "cuda"
         )
         model.to_empty(device=init_device)
-        model.init_weights()
+        if job_config.training.enable_cpu_offload:
+            with torch.device("cuda"):
```

Comment:
I somehow think the `model.init_weights(buffer_device="cuda")` change sounds better. It is straightforward on what we want to achieve, while making minimum change to code.